operations/configuration_management/Management_Node_Image_Customization.md

```diff
@@ -423,57 +426,80 @@ new customized image.
 
 1. (`ncn-mw#`) Update boot parameters for a management node.
 
-    Perform the following procedure **for each xname** identified in the previous step.
-
-    1. Get the existing `metal.server` setting for the component name (xname) of the node of interest.
+    1. When working with a group of `XNAMES` (use the previous step to collect the group of `XNAMES`).
 
         ```bash
-        XNAME=<node-xname>
-        METAL_SERVER=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' \
+        for XNAME in "${XNAMES[@]}"; do
+          echo "--- $XNAME ---"
+          METAL_SERVER=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' \
             | awk -F 'metal.server=' '{print $2}' \
             | awk -F ' ' '{print $1}')
-        echo "${METAL_SERVER}"
+          echo "current metal.server: ${METAL_SERVER}"
+          S3_ARTIFACT_PATH="boot-images/${NEW_IMS_IMAGE_ID}"
+          NEW_METAL_SERVER="s3://${S3_ARTIFACT_PATH}/rootfs"
+          echo "    new metal.server: ${NEW_METAL_SERVER}"
+          PARAMS=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' | \
+            sed "/metal.server/ s|${METAL_SERVER}|${NEW_METAL_SERVER}|" | \
+            sed "s/metal.no-wipe=1/metal.no-wipe=0/" | \
+            tr -d \")
+          cray bss bootparameters update --hosts "${XNAME}" \
+            --kernel "s3://${S3_ARTIFACT_PATH}/kernel" \
+            --initrd "s3://${S3_ARTIFACT_PATH}/initrd" \
+            --params "${PARAMS}"
+        done
         ```
```

Comment:
I don't see any return codes being checked in this code snippet. If it hits a problem on the first xname but the rest work fine, the error messages will be at the top of the output. An admin who only looks at the tail end of the output would possibly overlook the earlier failures. Ideally it should check for failing commands and break out of the loop immediately when one is hit, with an appropriate error message.

This may make more sense to be moved into a separate shell script file, for this reason.

Reply:
I did this rather impatiently, because I couldn't believe the steps were having me run this once-per NCN before. I can shell-script this.